rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3135,6 +3147,16 @@ extern "C" rmw_ret_t rmw_send_response(
   cdds_request_header_t header;
   memcpy(&header.guid, request_header->writer_guid, sizeof(header.guid));
   header.seq = request_header->sequence_number;
+  // if the number of writers matching our request reader equals the number
+  // of readers matching our response writer, we have a pretty decent claim
+  // to have discovered this client's response reader.
+  //
+  // it is pretty horrid ... but blocking the service is the only option if
+  // the client is unable to determine that it has been fully discovered by
+  // the service.
+  while (!check_for_response_reader(info->service.sub->enth, info->service.pub->enth)) {
+    dds_sleepfor(DDS_MSECS(10));
+  }
```

Comment:
@eboasson what if the client goes away before sending the response? I'm fine with the busy wait, but it should timeout at some point. Unless we can detect that the request writer went away.

Reply:
The DDS discovery will discover the disappearance of the client, remove all matches with it and decrement the `current_count`. So I believe that case is covered. (But as I remarked above, there is a problem if you create/delete clients all the time.)

Reply:
I see. I'm still a bit wary about indefinite waits. Even if it rarely occurs, to timeout and fail on replying would give calling code a chance to do something about it.